src/mlpack/methods/ann/dists/normal_distribution.hpp

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file normal_distribution.hpp
+ * @author xiaohong ji
+ *
+ * Definition of the Normal distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "../activation_functions/logistic_function.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Normal Distribution function.
+ *
+ * Normal distribution is a function which accepts a mean and a standard deviation
+ * term and creates a probablity distribution out of it.
+ */
+class NormalDistribution
+{
+ private:
+  //! Mean of the distribution.
+  arma::vec mean;
+  //! Standard deviation of the distribution.
+  arma::vec sigma;
+
+  // pi
+  static const constexpr double pi = 3.14159265358979323846264338327950288;
+
+ public:
+  /**
+   * Default constructor, which creates a Normal distribution with zero
+   * dimension.
+   */
+  NormalDistribution();
+
+  /**
+   * Create a Normal distribution with the given mean and sigma.
+   */
+  NormalDistribution(const arma::vec& mean, const arma::vec& sigma);
```

Comment:
Would be nice to comment on each parameter using `@param `.

Reply:
Oh.. completely missed it